src/pad/src/ICeWall.cpp

```diff
@@ -0,0 +1,1070 @@
+    }
+    if (!check_inst->isFixed()) {
+      continue;
+    }
+    const odb::Rect check_rect = check_inst->getBBox()->getBox();
+    if (inst_rect.overlaps(check_rect)) {
+      const double dbus = block->getDbUnitsPerMicron();
+      logger_->error(utl::PAD,
+                     1,
+                     "Unable to place {} ({}) at ({:.3f}um, {:.3f}um) - "
+                     "({:.3f}um, {:.3f}um) as it "
+                     "overlaps with {} ({})",
+                     inst->getName(),
+                     inst->getMaster()->getName(),
+                     inst_rect.xMin() / dbus,
+                     inst_rect.yMin() / dbus,
+                     inst_rect.xMax() / dbus,
+                     inst_rect.yMax() / dbus,
+                     check_inst->getName(),
+                     check_inst->getMaster()->getName());
+    }
+  }
+  inst->setPlacementStatus(odb::dbPlacementStatus::FIRM);
+}
+
+void ICeWall::placeFiller(const std::vector<odb::dbMaster*>& masters,
+                          odb::dbRow* row)
+{
+  auto* block = getBlock();
+  if (block == nullptr) {
+    return;
+  }
+
+  if (row == nullptr) {
+    return;
+  }
+  const double dbus = block->getDbUnitsPerMicron();
+
+  std::vector<odb::dbMaster*> fillers = masters;
+  // remove nullptrs
+  fillers.erase(std::remove(fillers.begin(), fillers.end(), nullptr),
+                fillers.end());
+  // sort by width
+  std::stable_sort(fillers.begin(),
+                   fillers.end(),
+                   [](odb::dbMaster* r, odb::dbMaster* l) -> bool {
+                     // sort biggest to smallest
+                     return r->getWidth() > l->getWidth();
+                   });
+
+  const odb::Rect rowbbox = row->getBBox();
+
+  using namespace boost::polygon::operators;
+  using Rectangle = boost::polygon::rectangle_data<int>;
+  using Polygon90 = boost::polygon::polygon_90_with_holes_data<int>;
+  using Polygon90Set = boost::polygon::polygon_90_set_data<int>;
+  using Pt = Polygon90::point_type;
+
+  std::vector<Polygon90> placed_io;
+  for (auto* inst : getPadInstsInRow(row)) {
+    const odb::Rect bbox = inst->getBBox()->getBox();
+    std::array<Pt, 4> pts;
+    if (row->getDirection() == odb::dbRowDir::HORIZONTAL) {
+      pts = {Pt(bbox.xMin(), rowbbox.yMin()),
+             Pt(bbox.xMax(), rowbbox.yMin()),
+             Pt(bbox.xMax(), rowbbox.yMax()),
+             Pt(bbox.xMin(), rowbbox.yMax())};
+    } else {
+      pts = {Pt(rowbbox.xMin(), bbox.yMin()),
+             Pt(rowbbox.xMax(), bbox.yMin()),
+             Pt(rowbbox.xMax(), bbox.yMax()),
+             Pt(rowbbox.xMin(), bbox.yMax())};
+    }
+
+    debugPrint(
+        logger_,
+        utl::PAD,
+        "Fill",
+        2,
+        "Instance in {} -> {} ({:.3f}um, {:.3f}um) -> ({:.3f}um, {:.3f}um)",
+        row->getName(),
+        inst->getName(),
+        pts[0].x() / dbus,
+        pts[0].y() / dbus,
+        pts[2].x() / dbus,
+        pts[2].x() / dbus);
+
+    Polygon90 poly;
```

Comment:
I don't think you really need polygon here as this a one dimensional problem (the cells are all the same height).   https://www.boost.org/doc/libs/1_81_0/libs/icl/doc/html/index.html would suffice

Reply:
okay